query/src/test/java/com/connexta/replication/query/QueryServiceToolsTest.java

```diff
@@ -0,0 +1,128 @@
+
+package com.connexta.replication.query;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.NodeAdapterFactory;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.data.SiteType;
+import com.connexta.replication.api.impl.NodeAdapters;
+import com.connexta.replication.api.persistence.FilterIndexManager;
+import com.connexta.replication.api.persistence.FilterManager;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.QueueBroker;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryServiceToolsTest {
+
+  @Mock private NodeAdapters nodeAdapterFactories;
+
+  @Mock private SiteManager siteManager;
+
+  @Mock private FilterManager filterManager;
+
+  @Mock private FilterIndexManager filterIndexManager;
+
+  @Mock private QueueBroker queueBroker;
+
+  private long period = 60L;
+
+  private QueryServiceTools queryServiceTools;
+
+  @Before
+  public void setUp() throws Exception {
+    queryServiceTools =
+        new QueryServiceTools(
+            nodeAdapterFactories,
+            siteManager,
+            filterManager,
+            filterIndexManager,
+            queueBroker,
+            period);
+  }
+
+  @Test
+  public void getPeriod() {
+    assertThat(queryServiceTools.getPeriod(), Matchers.is(period));
+  }
+
+  @Test
+  public void getPeriodDefault() {
+    queryServiceTools =
+        new QueryServiceTools(
+            nodeAdapterFactories, siteManager, filterManager, filterIndexManager, queueBroker, 0);
+    assertThat(queryServiceTools.getPeriod(), Matchers.is(TimeUnit.MINUTES.toSeconds(5)));
+  }
+
+  @Test
+  public void allSites() {
+    queryServiceTools.allSites();
+    verify(siteManager).objects();
+  }
+
+  @Test
+  public void getAdapterFor() {
+    Site site = mock(Site.class);
+    when(site.getType()).thenReturn(SiteType.DDF);
+    NodeAdapterFactory nodeAdapterFactory = mock(NodeAdapterFactory.class);
+    when(nodeAdapterFactories.factoryFor(any(SiteType.class))).thenReturn(nodeAdapterFactory);
+    queryServiceTools.getAdapterFor(site);
+    verify(nodeAdapterFactory).create(null);
+  }
+
+  @Test
+  public void activeFiltersFor() {
+    Filter filter = mock(Filter.class);
+    Site site = mock(Site.class);
+    when(filterManager.filtersForSite(null)).thenReturn(Stream.of(filter));
+    assertThat(queryServiceTools.activeFiltersFor(site).findFirst().get(), Matchers.is(filter));
```

Comment:
✏️ Although yu are verifying you are getting `filter` back, you are not verifying that this is the only thing you are getting back. I would suggest you collect the result into a list and then use `Matchers.contains()` or `Matchers.containsInAnyorder()` to verify the content is exactly what you expect and nothing more